src/main/kotlin/de/itemis/mps/gradle/runmigrations/Plugin.kt

```diff
@@ -0,0 +1,83 @@
+package de.itemis.mps.gradle.runmigrations
+
+import de.itemis.mps.gradle.*
+import net.swiftzer.semver.SemVer
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Copy
+import org.gradle.kotlin.dsl.withGroovyBuilder
+import java.io.File
+import javax.inject.Inject
+
+open class MigrationExecutorPluginExtensions @Inject constructor(of: ObjectFactory, project: Project) : BasePluginExtensions(of, project) {
+    val force: Property<Boolean> = of.property(Boolean::class.java).convention(false)
+}
+@Suppress("unused")
+open class RunMigrationsMpsProjectPlugin : Plugin<Project> {
+    override fun apply(project: Project) {
+        project.run {
+            val extension = extensions.create("runMigrations", MigrationExecutorPluginExtensions::class.java)
+            afterEvaluate {
+                val mpsLocation = extension.mpsLocation.convention{ File(project.buildDir, "mps") }.map { it.asFile }.get()
+                val projectLocation = extension.projectLocation.getOrElse { throw GradleException("No project path set") }
+                if(!file(projectLocation).exists()) {
+                    throw GradleException("The path to the project doesn't exist:$projectLocation")
+                }
+                val forceMigration = extension.force
+                
+                val mpsVersion = extension.getMPSVersion()
```

Comment:
How will this part work with a custom RCP? Will I have to know which version of my RCP corresponds to which version of MPS? I'd rather not to. What happens if we don't do the check and a user uses 'force' on an early version of MPS?

Reply:
Explanation for others: as long as `mpsLocation` and `mpsVersion` are set, it will work. `mpsVersion` can be set to a random value like 999.999. The only requirement is that [ semantic versioning](https://semver.org/) is used for the version. The MPS version checks should pass this way.